WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+import { Button } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import UserBadge from '../UserBadge';
+
+import '../../stylesheets/delegates/style.scss';
+
+const dasherize = (string) => string.replace(/_/g, '-');
+
+export default function Delegates({
+  delegates,
+  isEditVisible,
+}) {
+  const seniorDelegates = React.useMemo(() => delegates
+    .filter((user) => user.delegate_status === 'senior_delegate')
+    .sort((user1, user2) => {
+      if (user1.region < user2.region) {
+        return -1;
+      } if (user1.region > user2.region) {
+        return 1;
+      }
+      return 0;
+    }), [delegates]);
+
+  // TO_VERIFY: I assume there are no cases where there are no delegates without
+  // senior delegate unless they are senior delegate
```

Comment:
I assume this still needs to be resolved before the PR is ready to be merged?

Reply:
Hi Kevin, Sorry for late reply. Actually yes. As far as I know, there are no cases right now where there are delegates without senior delegates (unless they themself is a senior delegate). So, it should be safe to assume this, but for a double verification, would like to get a yes from anybody from Board or our Leader (Gregor).

Reply:
It is worth noting that the old Ruby implementation had a `@delegates_without_senior_delegates` variable. So it does seem like this edge case can occur.

Why is it so important to explicitly assume that _Senior Delegates_ are the top of the hierarchy? I'd like to suggest some approach where you iterate over the list of delegates and group them by whoever seems to be the "senior-most Delegate"? In essence, just interpret `senior_delegate_id` as a hierarchy that you can traverse upwards

Reply:
Actually, I feel earlier there were cases where delegates existed without senior delegates. But now, there are nobody like that. Even though I'm 99% sure that the case doesn't exist now, I would like to get a confirmation about the same. :-)

The next PR coming immediately after merging this will be introducing of regions table. Once that comes, every delegates will be mapped to the region instead of senior delegate and this issue won't be there anymore.

This PR was basically to implement the same thing that was there earlier, and I feel adding @delegates_without_senior_delegates won't be so good because we don't have any use-case for that right now and once the next PR comes, the code has to be removed.

What do you think regarding the same?

Reply:
I think I didn't give much context regarding regions table. It's the same regions table mentioned in [this PR](https://github.com/thewca/worldcubeassociation.org/pull/5417). I was initially thinking to do both delegates table and regions table together, but there are plenty of cases to handle in that case and is completely out of my scope. So I thought doing it step by step will be better. The next PR will have just the regions table implementation (it won't be dead table for sure, I'll be connecting the delegates to regions instead of senior delegates in the same PR). Some PRs after that will be some UI improvements to make the delegates table implementation easy.  And finally, I'll be bringing the delegates table PR.

Reply:
Makes sense to me!

Reply:
Thanks Greg.